### ISSUE TITLE
RegistryKeyLookupFailed `CoderModulesPath' @ error/module.c/GetMagickModulePath/665.

### ISSUE THREAD

**Issue by Watson1978** [ImageMagick/ImageMagick6#231]
### ImageMagick version

 6.9.12-84 Q16-HDRI x64

### Operating system

Windows

### Operating system, version and so on

Windows 11 22H2

### Description

convert command causes `RegistryKeyLookupFailed` error when resize image.

### Steps to Reproduce

```
C:\tmp>echo %PATH%
C:\Program Files\ImageMagick-6.9.12-Q16-HDRI;;;;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\gs\gs10.01.1\bin;C:\Ruby31-x64\bin;C:\Ruby31-x64\msys64\ucrt64\bin;C:\Ruby31-x64\msys64\usr\bin;C:\Users\watso\AppData\Local\Microsoft\WindowsApps;C:\Users\watso\AppData\Local\GitHubDesktop\bin;C:\Users\watso\AppData\Local\Programs\Microsoft VS Code\bin

C:\tmp>identify -version
Version: ImageMagick 6.9.12-84 Q16-HDRI x64 0731a37:20230402 https://legacy.imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 193431943
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zlib

C:\tmp>convert -resize 300x200 sample.jpg out.gif
convert.exe: RegistryKeyLookupFailed `CoderModulesPath' @ error/module.c/GetMagickModulePath/665.
convert.exe: no decode delegate for this image format `JPEG' @ error/constitute.c/ReadImage/581.
convert.exe: no images defined `out.gif' @ error/convert.c/ConvertImageCommand/3229.

C:\tmp>convert -resize 300x200 sample.jpg test.jpg
convert.exe: RegistryKeyLookupFailed `CoderModulesPath' @ error/module.c/GetMagickModulePath/665.
convert.exe: no decode delegate for this image format `JPEG' @ error/constitute.c/ReadImage/581.
convert.exe: no images defined `test.jpg' @ error/convert.c/ConvertImageCommand/3229.
```

**Comment by dlemstra** [comment]
This looks like an issue with how you installed ImageMagick? It is unable to find that path where the modules can be found in the registry.

**Comment by Watson1978** [comment]
When I installed, I selected `Install development headers and librarries for C and C++` in `Select Additional Tasks` of installer. (Ref. https://github.com/rmagick/rmagick#windows)

**Comment by merrilymeredith** [comment]
I'm seeing this also, but 6.9.12-84 Q16 _x86_.  The installer's registry keys all look normal; I didn't do anything special during install, just uninstalled a prior version then clicked through this installer, and also included headers.

There's no change if I set MAGICK_HOME to give it an extra hint, then if I set MAGICK_CODER_MODULE_PATH it's satisfied with that path but I still get `... no decode delegate ...` and nothing returned by `identify -list format`.

**Comment by dlemstra** [comment]
Thanks for the info @merrilymeredith. I have applied a patch that should fix this in the next release.

**Comment by Watson1978** [comment]
Thanks.
This issue was solved by ImageMagick 6.9.12-85

![image](https://user-images.githubusercontent.com/199156/232533203-18faa284-62e1-430c-997b-d7a35ba5322c.png)
